src/index.ts

```diff
@@ -385,8 +396,15 @@ export async function cli(args: string[]) {
     namedExports: undefined,
     webDependencies: undefined,
   };
-  const doesWhitelistExist = !!webDependencies;
-  const arrayOfDeps = webDependencies || Object.keys(pkgManifest.dependencies || {});
+
+  let doesWhitelistExist = true;
+  const {dependencies, dependencyTree} = scanImports(entry || '', {dest});
+  const arrayOfDeps = [...webDependencies, ...dependencies];
+  if (!arrayOfDeps.length) {
+    doesWhitelistExist = false;
+    arrayOfDeps.push(...Object.keys(pkgManifest.dependencies || {}));
+  }
```

Comment:
Cleaned up src/index significantly with the suggestions from earlier!

Reply:
nice!